notebooks/3.3_Quri_Parts.ipynb

```diff
@@ -0,0 +1,381 @@
+      
+      
+     ]
+    }
+   ],
+   "source": [
+    "print(\"Qubit count:\", circuit.qubit_count) \n",
+    "print(\"Circuit depth:\", circuit.depth)\n",
+    "\n",
+    "gates = circuit.gates # .gates returns the gates in the circuit as a sequence \n",
+    "print(\"# of gates in the circuit:\", len(gates))",
+    "for gate in gates: \n",
+    "print(gate)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "`QuantumCircuit` objects with the same number of qubits can be combined and extended:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Combined circuit: (QuantumGate(name= X , target_indices=(0,), control_indices=(), params=(), pauli_ids=()), QuantumGate(name= X , target_indices=(0,), control_indices=(), params=(), pauli_ids=()), QuantumGate(name= RX, target_indices=(1,), control_indices=(), params=(1.0471975511965976,), pauli_ids=()), QuantumGate(name= CNOT, target_indices=(1,), control_indices=(2,), params=(), pauli_ids=()), QuantumGate(name= PauliRotation, target_indices=(0, 1, 2), control_indices=(), params=(1.0471975511965976,), pauli_ids=(1, 2, 3)), QuantumGate(name='Y', target_indices=(1,), control_indices=(), params=(), pauli_ids=()), QuantumGate(name= H, target_indices=(2,), control_indices=(), params=(), pauli_ids=()))\n",
+      "Extended circuit: (QuantumGate(name= Y, target_indices=(1,), control_indices=(), params=(), pauli_ids=()), QuantumGate(name= H, target_indices=(2,), control_indices=(), params=(), pauli_ids=()), QuantumGate(name= X, target_indices=(0,), control_indices=(), params=(), pauli_ids=()), QuantumGate(name= X, target_indices=(0,), control_indices=(), params=(), pauli_ids=()), QuantumGate(name= RX, target_indices=(1,), control_indices=(), params=(1.0471975511965976,), pauli_ids=()), QuantumGate(name= CNOT, target_indices=(1,), control_indices=(2,), params=(), pauli_ids=()), QuantumGate(name= PauliRotation , target_indices=(0, 1, 2), control_indices=(), params=(1.0471975511965976,), pauli_ids=(1, 2, 3)))"
+     ]
+    }],
+   "source": [
+    "circuit2 = QuantumCircuit(3)\n",
+    "circuit2.add_Y_gate(1)\n",
+    "circuit2.add_H_gate(2)\n",
+    "\n",
+    "combined = circuit + circuit2 # equivalent: combined = circuit.combine(circuit2) \n",
+    "print(\"Combined circuit:\", combined.gates) \n",
+    "\n",
+    "circuit2 += circuit # equivalent: circuit2.extend(circuit) \n",
+    "print(\"Extended circuit:\", circuit2.gates)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Operators:\n",
+    "An operator is defined as a set of pairs of a Pauli label and its complex coefficient. For example, an operator:<br>",
+    "$(0.5+0.5j)*X0 Y1 + 0.2j*Z0 Z2 + (0.3+0.4j)*I$",
```

Comment:
I think it would be more aesthetically pleasing if we use subscripts for the qubit index in the equation.